### PR TITLE
refactor: how it works

### PR DESCRIPTION
**Pull request by simboonlong**
to refactor further from [discussion](https://github.com/TradeTrust/tradetrust-website/pull/581#discussion_r894174674)

- type not typed
- types move to local component level
- rename variables

### CODE REVIEW

File: src/components/HomePageContent/HowItWorksSection/DocumentContent.tsx [hunk 1]
```diff
@@ -1,23 +1,355 @@
...
-    message: "See how things change from the use of TradeTrust-enabled solutions when dealing with eBL ",
-    users: TransferableRecordPersonas,
+    type: DocumentType.TRANSFERABLE_RECORD,
+    text: {
+      description:
+        "Transferable Records are trade documents that entitle the holder to claim the performance of an obligation or ownership.",
+      examples:
+        "Examples of transferable records typically include bills of lading, bills of exchange, promissory notes, and warehouse receipts.",
+      message: "See how things change from the use of TradeTrust-enabled solutions when dealing with eBL ",
+    },
```
Comment (simboonlong) [inline src/components/HomePageContent/HowItWorksSection/DocumentContent.tsx:12]: what was `examples`? group them as text related fields

File: src/components/HomePageContent/HowItWorksSection/PersonaModal.tsx [hunk 2]
```diff
@@ -31,44 +41,40 @@ export const PersonaModal: FunctionComponent<PersonaProps> = ({ personaIndex, de
...
-                  <div className={`lg:inline ${content === selectedContentType ? "inline" : "hidden"}`}>
-                    <Steps contentType={content} stepsDetails={content === ContentType.THEN ? thenSteps : nowSteps} />
-                  </div>
+              <>
+                <ProcessLegends />
+                <div className="flex flex-row justify-center lg:hidden">
+                  {contentType.map((content, index) => {
+                    const cssState = content === selectedContentType ? "font-gilroy-bold underline" : "font-normal";
+                    const cssAlign = content === ContentType.THEN ? "mr-5" : "";
+                    return (
```
Comment (simboonlong) [inline src/components/HomePageContent/HowItWorksSection/PersonaModal.tsx:53]: separate alignment-related class from state active css classes

File: src/components/HomePageContent/HowItWorksSection/PersonaModal.tsx [hunk 3]
```diff
@@ -31,44 +41,40 @@ export const PersonaModal: FunctionComponent<PersonaProps> = ({ personaIndex, de
...
-                    onClick={() => setSelectedContentType(content)}
-                  >
-                    {content}
-                  </h3>
-                ))}
-              </div>
-            )}
-            {thenSteps &&
-              nowSteps &&
-              contentType.map((content, index) => (
```
Comment (simboonlong) [inline src/components/HomePageContent/HowItWorksSection/PersonaModal.tsx:65]: remove duplicate condition

File: src/components/HomePageContent/HowItWorksSection/Steps.tsx [hunk 4]
```diff
@@ -1,44 +1,48 @@
-import React from "react";
-import { ContentType } from "../../../types";
+import React, { FunctionComponent } from "react";
+import { StepDetail, StepsProps, ContentType } from "./types";
 
-export const Steps: React.FunctionComponent<any> = ({ contentType, stepsDetails }) => {
-  const keyValueBorderStyle: { [key: string]: string } = {
```
Comment (simboonlong) [inline src/components/HomePageContent/HowItWorksSection/Steps.tsx:5]: type them

File: src/components/HomePageContent/HowItWorksSection/Steps.tsx [hunk 5]
```diff
@@ -1,44 +1,48 @@
...
     THEN: "border-dashed",
     NOW: "border-solid",
   };
+
   return (
     <div className="flex flex-col justify-center">
-      {(contentType === ContentType.THEN || contentType === ContentType.NOW) && (
+      {contentType !== ContentType.BENEFIT && (
         <h3 className="hidden text-lemon-500 text-center mt-12 lg:inline">{contentType}</h3>
       )}
```
Comment (simboonlong) [inline src/components/HomePageContent/HowItWorksSection/Steps.tsx:38]: previously already has weird side effect from data content, ignoring this

File: src/components/HomePageContent/HowItWorksSection/Steps.tsx [hunk 6]
```diff
@@ -1,44 +1,48 @@
...
       <div className="flex flex-wrap m-auto lg:px-16 2xl:px-0 items-start lg:w-full 2xl:max-w-7xl">
-        {stepsDetails.map((steps: any, index: any) => (
+        {stepsDetails.map((stepDetail: StepDetail, index: number) => (
           <React.Fragment key={index}>
```
Comment (simboonlong) [inline src/components/HomePageContent/HowItWorksSection/Steps.tsx:41]: type them

File: src/components/HomePageContent/HowItWorksSection/index.tsx [hunk 7]
```diff
@@ -1,45 +1,36 @@
...
 import { DocumentContent } from "./DocumentContent";
-import { DocumentTypeContent } from "../../../types";
-import { DocumentTypeDetails } from "./DocumentTypeDetails";
+import { DocumentType, DocumentTypeContent, Persona } from "./types";
+import { PersonaCard } from "./PersonaCard";
 
-enum DocumentType {
-  VERIFIABLE_DOCUMENT = "Verifiable Documents",
-  TRANSFERABLE_RECORD = "Transferable Records",
-}
```
Comment (simboonlong) [inline src/components/HomePageContent/HowItWorksSection/index.tsx:10]: moved them to reuse

File: src/components/HomePageContent/HowItWorksSection/index.tsx [hunk 8]
```diff
@@ -1,45 +1,36 @@
...
+};
 
 export const HowItWorksSection: FunctionComponent = () => {
   const [selectedDocumentTypeLabel, setSelectedDocumentTypeLabel] = useState<DocumentType>(
     DocumentType.TRANSFERABLE_RECORD
   );
-  const documentType: DocumentType[] = [DocumentType.TRANSFERABLE_RECORD, DocumentType.VERIFIABLE_DOCUMENT];
-  const [selectedDocumentTypeContent, setSelectedDocumentTypeContent] = useState<DocumentTypeContent>(
-    DocumentContent[0]
-  );
```
Comment (simboonlong) [inline src/components/HomePageContent/HowItWorksSection/index.tsx:19]: redundant

File: src/types/index.tsx [hunk 9]
```diff
@@ -30,51 +30,6 @@ export type Resource = {
   date?: string;
 };
 
-export interface PersonaProps {
-  personaIndex: number;
-  details: Persona;
-}
```
Comment (simboonlong) [inline src/types/index.tsx:36]: moved all these types to local component. should that component no longer relevant, can delete the folder entirely

File: src/components/HomePageContent/HowItWorksSection/Steps.tsx [hunk 10]
```diff
@@ -1,44 +1,48 @@
...
-            <div
```
Comment (simboonlong) [inline src/components/HomePageContent/HowItWorksSection/Steps.tsx:18]: typed them

File: src/types/index.tsx (hunk 9 above)
Comment (isaackps) [reply src/types/index.tsx:36]: ok

Comment (isaackps) [reply src/types/index.tsx:36]: ok

File: src/components/HomePageContent/HowItWorksSection/index.tsx (hunk 8 above)
Comment (isaackps) [reply src/components/HomePageContent/HowItWorksSection/index.tsx:19]: yup ok

File: src/components/HomePageContent/HowItWorksSection/index.tsx (hunk 7 above)
Comment (isaackps) [reply src/components/HomePageContent/HowItWorksSection/index.tsx:10]: ok

File: src/components/HomePageContent/HowItWorksSection/PersonaModal.tsx (hunk 2 above)
Comment (isaackps) [reply src/components/HomePageContent/HowItWorksSection/PersonaModal.tsx:53]: ok

File: src/components/HomePageContent/HowItWorksSection/Steps.tsx (hunk 4 above)
Comment (isaackps) [reply src/components/HomePageContent/HowItWorksSection/Steps.tsx:5]: ok

File: src/components/HomePageContent/HowItWorksSection/Steps.tsx (hunk 5 above)
Comment (isaackps) [reply src/components/HomePageContent/HowItWorksSection/Steps.tsx:38]: ok

File: src/components/HomePageContent/HowItWorksSection/Steps.tsx (hunk 10 above)
Comment (isaackps) [reply src/components/HomePageContent/HowItWorksSection/Steps.tsx:18]: ok

**Review by isaackps — APPROVED** [review]
LGTM

**Pull request merged**